torch/_dynamo/eval_frame.py

```diff
@@ -789,6 +793,75 @@ def run_node(self, n):
                 r.node.meta["val"] = self.current_node.meta["val"]
             return r
 
+    class AddRuntimeAssertsInInputConstraint(torch.fx.interpreter.Transformer):
+        def __init__(
+            self,
+            m,
+        ):
+            super().__init__(m)
+            self.count = 0
+            self.constraints_id_to_constraint = defaultdict(list)
+            if constraints is not None:
+                # map input tensor to it's constraints
+                for constraint in constraints:
+                    self.constraints_id_to_constraint[constraint.t_id].append(constraint)
+
+        def placeholder(self, target, args, kwargs):
+            arg = super().placeholder(target, args, kwargs)
+            orig_inp_id = id(flat_args[self.count])
+
+            if orig_inp_id not in self.constraints_id_to_constraint:
+                self.count += 1
+                return arg
+
+            constraints = self.constraints_id_to_constraint[orig_inp_id]
+
+            # Convert simple sympy Integers into concrete int to
+            # insert into graph
+            def _convert_to_int(val):
+                if val == sympy.oo:
+                    return math.inf
+                if isinstance(val, sympy.Integer):
+                    return int(val)
+                # TODO ignore expressions for now
+                return None
+
+            for constraint in constraints:
+                constraint_range = constraint.constraint_range
+                if constraint_range is None:
+                    continue
+
+                min_int_val = _convert_to_int(constraint_range.vr.lower)
+                max_int_val = _convert_to_int(constraint_range.vr.upper)
+
+                if min_int_val is None and max_int_val is None:
```

Comment:
I don't think these can be None? You should be able to trust upstream and assert in _convert_to_int I think...

Reply:
It will be None if the expression is not something we can convert to integer (complicated guard expression).